File: Difficulty: Easy/Second Largest/second-largest.cpp
class Solution {
  public:
    int getSecondLargest(vector<int> &arr) {
        // code here
        int n = arr.size();
        if(n < 2){
            return -1;
        }
        int secondMax = INT_MIN;
        int firstMax = INT_MIN + 1;
        for(int i =0; i<n; i++){
            firstMax = max(firstMax, arr[i]);
        }
        for(int i =0; i<n; i++){
            if(arr[i]<firstMax && arr[i]> secondMax){
                secondMax = arr[i];
            }
        }
        if(secondMax != INT_MIN){
            return secondMax;
        }
        else{
            return -1;
        }
    }
};

File: Difficulty: Easy/Second Largest/second-largest.py
class Solution:
    def getSecondLargest(self, arr):
        lar = -1
        slr = -1

        for num in arr:
            if num > lar:
                slr = lar
                lar = num
            elif num > slr and num != lar:
                slr = num

        return slr